app/jobs/google/search_keyword_job.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Google
+  class SearchKeywordJob < ApplicationJob
+    queue_as :default
+
+    def perform(keyword)
+      search_result = Google::SearchService.new(keyword[:name]).call
+      raise Google::Errors::SearchKeywordError unless search_result
+
+      update_keyword(keyword, search_result)
```

Comment:
Let reduce the magic string by moving it as a constant of the `Google::SearchService`
The pass it to the update method, instead of querying it there. For example, if we have another search service `Bing` then it easier to modify. 

```suggestion
      google_source = Source.find_or_create_by({ name: Google::SearchService::NAME })
      
      update_keyword(keyword, search_result, google_source)
```

Reply:
@byhbt I am not sure what you mean by "moving it as a constant of the Google::SearchService" but the benefit of "have another search service Bing then it easier to modify" sounds good, could you elaborate more on the example 🙏

Reply:
Yup, here is the idea:

This is `SearchKeywordJob` which calls the `Google::SearchService`. However, the limitation is in the `update_keyword`, the `source` we `find_or_create_by` is `Google`.

If one day, we don't want to search by the `Google::SearchService`, we would like to use `Bing::SearchService` instead, then we need to go to the `update_keyword`, update the string `{ name: 'Bing' }` there as well. 

So instead of that, the `update_keyword` method should care about what search are using, they just accept one more param `source` `update_keyword(keyword, search_result, google_source)`. So it could be easier to change right? 💭  We don't need to update the `Google` word multiple places.

Reply:
I moved the `source_name` to controller level so that `Keyword.callback action` knows which source to call. So this file had changed somewhat 🙏